src/lro/passthroughStrategy.ts

```diff
@@ -0,0 +1,23 @@
+import { LROStrategy, BaseResult, LROOperationStep } from "./models";
+import { SendOperationFn } from "./lroPoller";
+
+/**
+ * Creates a polling strategy based on BodyPolling which uses the provisioning state
+ * from the result to determine the current operation state
+ */
+export function createPassthroughStrategy<TResult extends BaseResult>(
+  initialOperation: LROOperationStep<TResult>
+): LROStrategy<TResult> {
+  return {
+    isTerminal: () => {
+      return true;
+    },
+    sendFinalRequest: () => {
+      // BodyPolling doesn't require a final get so return the lastOperation
+      return Promise.resolve(initialOperation);
+    },
+    poll: async () => {
+      throw new Error("Passthrough strategy should never poll");
```

Comment:
I'm a little fuzzy on this part. Does `poll` get called as a result of a method invocation the user can make? I'm wondering if it makes sense to have `poll` return the initialOperation as well, or be a no-op, rather than throwing an error.

Reply:
This poll function is internal to our LRO handling, users cannot call this directly, we'll decide if it is called or not in the update method in operation.ts. 

Core-LRO provides an interface with a poll function which is different to this one, core-lro poll would work with the "operation" state, it checks if the operation is complete by looking at the state.isCompleted, if it is completed it does nothing. If it is not completed, then it calls the **update** method we define. There we decide based on the strategy what needs to happen.

Passthrough strategy isTerminal will always be true so no poll should ever be called, I think that if we return the initial operation we may risk running into an infinite loop if a bug is introduced in passtrough's isTerminal and returns false